editoast/src/views/timetable/train_schedule.rs

```diff
@@ -590,6 +609,309 @@ async fn occupancy_blocks(
+/// If not, it infers the position and time based on track offsets and interpolation.
+fn compute_track_occupancy(
+    path_item_id: Option<NonBlankString>,
+    path_item_positions: &[u64],
+    track_ranges: &Vec<TrackRange>,
+    operational_point_track_offsets: &[TrackOffset],
+    report_train: &ReportTrain,
+    train_schedule: &models::TrainSchedule,
+) -> Option<(String, TrackOccupancy)> {
+    let path_projection = PathProjection::new(track_ranges);
+
+    if let Some(path_item_id) = path_item_id {
+        // If the path item match the op_id, checks if a stop duration exists
+        let schedule_item = train_schedule
+            .schedule
+            .iter()
+            .find(|schedule| schedule.at == path_item_id);
+        // Stop time at the operational point:
+        // Retrieve the stop time if it exists; otherwise, default to 0
+        let stop_duration = match schedule_item {
+            Some(schedule_item) => match schedule_item.stop_for.clone() {
+                Some(stop_for) => stop_for,
+                None => Duration::milliseconds(0).try_into().unwrap(),
+            },
+            // When the path item is the departure position without being a schedule point
+            None => Duration::milliseconds(0).try_into().unwrap(),
+        };
+
+        // Get the index of the path item in the list of path items that corresponds to the operational point
+        let index = train_schedule
+            .path
+            .iter()
+            .position(|path| path.id == path_item_id)
+            .expect("Unexcpected path item");
+        let position = path_item_positions[index];
+        let time = report_train.path_item_times[index];
+
+        // Retrieve the track section that matches the path item's position along the path
+        let track_section = match path_projection.get_location(position) {
+            TrackLocationFromPath::One(track_offset) => operational_point_track_offsets
+                .iter()
+                .find(|to| *to == &track_offset)
+                .expect("Invalid track offset")
+                .track
+                .clone(),
+            TrackLocationFromPath::Two(track_offset, track_offset1) => {
+                operational_point_track_offsets
+                    .iter()
+                    .find(|to| *to == &track_offset)
+                    .or(operational_point_track_offsets
+                        .iter()
+                        .find(|to| *to == &track_offset1))
+                    .expect("Invalid track offset")
+                    .track
+                    .clone()
+            }
+        };
+
+        Some((
+            track_section.to_string(),
+            TrackOccupancy {
+                time_begin: train_schedule.start_time + Duration::milliseconds(time as i64),
+                duration: stop_duration,
+                train_schedule_id: train_schedule.id,
+            },
+        ))
+    } else {
```

Comment:
We could return here to avoid indentation.

```suggestion
        return Some((
            track_section.to_string(),
            TrackOccupancy {
                time_begin: train_schedule.start_time + Duration::milliseconds(time as i64),
                duration: stop_duration,
                train_schedule_id: train_schedule.id,
            },
        ))
    }
```